functions/db_bootstrapper/run_image.sh

```
@@ -10,22 +10,79 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
+# Color codes for terminal output
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[1;33m'
+BLUE='\033[0;34m'
+PURPLE='\033[0;35m'
+CYAN='\033[0;36m'
+NC='\033[0m' # No Color
+
+# Helper functions for colored output
+print_header() {
+    echo ""
+    echo "${BLUE}=== $1 ===${NC}"
+}
+
+print_success() {
+    echo "${GREEN}✓ $1${NC}"
+}
+
+print_warning() {
+    echo "${YELLOW}⚠ $1${NC}"
+}
+
+print_error() {
+    echo "${RED}✗ $1${NC}"
+}
+
+print_info() {
+    echo "${CYAN}ℹ $1${NC}"
+}
+
+print_step() {
+    echo "→ $1"
+}
+
+# Function to run commands with progress indication
+run_with_progress() {
+    local message="$1"
+    local command="$2"
+    
+    print_step "$message"
+    if eval "$command" > /dev/null 2>&1; then
+        print_success "$message completed"
+        return 0
+    else
+        print_error "$message failed"
+        return 1
+    fi
+}
+
 NOCACHE=0
-DEBUG=0
 PORT=9001
 DATA_FILE=""
 
+HELP_MESSAGE="
+Usage: $0 [options]
+
+Options:
+    --no-cache: Build without cache (default: false)
+    --data-file: Path to the data file to upload to S3 (default: "")
+    --port: Port to use (default: 9001)
+    --help: Show this help message
+"
+
+print_header "Database Bootstrapper Test Environment Setup"
+
 # parse arguments
 while [[ $# -gt 0 ]]; do
     case $1 in
         --no-cache)
             NOCACHE=1
             shift
             ;;
-        --debug)
-            DEBUG=1
-            shift
-            ;;
         --data-file)
```

Fail fast if the file isn’t there.

```bash
if [ -n "${DATA_FILE}" ]; then
  [ -f "${DATA_FILE}" ] || { print_error "Data file not found: ${DATA_FILE}"; exit 1; }
  ...
fi
```